src/runtime_src/core/edge/user/shim.cpp

```diff
@@ -371,6 +373,37 @@ int ZYNQShim::xclSyncBO(unsigned int boHandle, xclBOSyncDirection dir, size_t si
   return ioctl(mKernelFD, DRM_IOCTL_ZOCL_SYNC_BO, &syncInfo);
 }
 
+int ZYNQShim::xclCopyBO(unsigned int dst_boHandle, unsigned int src_boHandle, size_t size,
+                        size_t dst_offset, size_t src_offset)
+{
+    int ret = -EOPNOTSUPP;
+#ifdef __aarch64__
+    xrt_core::cmd_bo bo = mCmdBOCache->alloc();
+    ert_fill_copybo_cmd(bo.second, src_boHandle, dst_boHandle,
+                        src_offset, dst_offset, size);
+
+    ret = xclExecBuf(bo.first);
+    if (ret) {
+        mCmdBOCache->release(bo);
```

Comment:
I don't think we should expose the concept of a command bo cache inside functions that use / allocate command BOs.  The cache is a implementation detail to hide within existing xclAlloc/Free APis.